pulp_container/app/registry.py

```diff
@@ -117,7 +130,67 @@ async def get_tag(self, request):
                 pk__in=await sync_to_async(repository_version.get_content)(), name=tag_name
             )
         except ObjectDoesNotExist:
-            raise PathNotResolved(tag_name)
+            if distribution.remote:
+                remote = await distribution.remote.acast()
+
+                relative_url = "/v2/{name}/manifests/{tag}".format(
+                    name=remote.namespaced_upstream_name, tag=tag_name
+                )
+                tag_url = urljoin(remote.url, relative_url)
+                downloader = remote.get_in_memory_downloader(url=tag_url)
+                response = await downloader.run(extra_data={"headers": V2_ACCEPT_HEADERS})
+
+                set_guid(generate_guid())
+                task = await sync_to_async(dispatch)(
+                    download_image_data,
+                    exclusive_resources=[repository_version.repository],
+                    kwargs={
+                        "repository_pk": repository_version.repository.pk,
+                        "remote_pk": remote.pk,
+                        "tag_name": tag_name,
+                        "response_data": response.data,
+                    },
+                )
+
+                # waiting shortly for the task to be completed since a container client could
+                # request related content (i.e., manifests and blobs) and halt the pull operation
+                # if the content was not immediately served
+                for dummy in range(3):
+                    time.sleep(1)
+                    task = await Task.objects.aget(pk=task.pk)
+                    if task.state == "completed":
+                        await task.adelete()
+                        break
+                    elif task.state in ["waiting", "running"]:
+                        continue
+                    else:
+                        error = task.error
+                        await task.adelete()
+                        raise Exception(str(error))
+                else:
+                    raise HTTPTooManyRequests()
+
+                try:
+                    manifest_data = json.loads(response.data)
+                except json.decoder.JSONDecodeError:
+                    raise PathNotResolved(tag_name)
+                else:
+                    encoded_data = response.data.encode("utf-8")
+                    digest = calculate_digest(encoded_data)
```

Comment:
can you use reponse.artifact_arrtibutes?

Reply:
Nice idea.